packages/sdk/client-services/src/packlets/logging/logging-service.ts

```diff
@@ -0,0 +1,55 @@
+//
+// Copyright 2022 DXOS.org
+//
+
+import { Event } from '@dxos/async';
+import { Stream } from '@dxos/codec-protobuf';
+import { log, LogEntry as NaturalLogEntry, LogProcessor, shouldLog } from '@dxos/log';
+import { LoggingService, LogEntry, QueryLogsRequest } from '@dxos/protocols/proto/dxos/client/services';
+
+/**
+ * Logging service used to spy on logs of the host.
+ */
+export class LoggingServiceImpl implements LoggingService {
+  private readonly _active = new Set<Event<LogEntry>>();
+
+  constructor() {
+    log.runtimeConfig.processors.push(this._logProcessor);
+  }
+
+  queryLogs(request: QueryLogsRequest): Stream<LogEntry> {
+    const logs = new Event<LogEntry>();
+    this._active.add(logs);
+    return new Stream<LogEntry>(({ ctx, next }) => {
+      const handler = (entry: LogEntry) => {
+        if (shouldLog(transformLogEntry(entry), request.filters)) {
```

Comment:
wouldn't it go the other way: you're getting NaturalLogEntry from the logger and converting it to the protobuf LogEntry to stream it